lld/ELF/Relocations.cpp

```diff
@@ -1111,7 +1125,19 @@ void RelocationScanner::processAux(RelExpr expr, RelType type, uint64_t offset,
     } else if (!sym.isTls() || ctx.arg.emachine != EM_LOONGARCH) {
       // Many LoongArch TLS relocs reuse the R_LOONGARCH_GOT type, in which
       // case the NEEDS_GOT flag shouldn't get set.
-      sym.setFlags(NEEDS_GOT);
+      bool needsGotAuth =
+          (expr == R_AARCH64_AUTH_GOT || expr == R_AARCH64_AUTH_GOT_PAGE_PC);
+      uint16_t flags = sym.flags.load(std::memory_order_relaxed);
+      if (!(flags & NEEDS_GOT)) {
+        if (needsGotAuth)
+          sym.setFlags(NEEDS_GOT | NEEDS_GOT_AUTH);
+        else
+          sym.setFlags(NEEDS_GOT);
```

Comment:
```suggestion
       uint16_t flag = needsGotAuth? NEEDS_GOT | NEEDS_GOT_AUTH :  NEEDS_GOT;
       sym.setFlags(flag);
```
Maybe this? Or alternativly:
```suggestion
       uint16_t flag = NEEDS_GOT;
        if (needsGotAuth)
          flag |= NEEDS_GOT_AUTH;
        sym.setFlags(flag);
```

Not a big improvement(in any) in either case, but I always find it easier to limit the control flow to just setting the flag.

Reply:
Thanks for suggestion, I've applied the approach with ternary operator, but w/o using the temp flag variable (just put the ternary inside `setFlags` invocation) - this fits one line and looks readable enough IMHO. See 262b4c6eb0e655e3b47350e5f390d4997d7d0431